.circleci/config.yml

```diff
@@ -30,13 +39,63 @@ jobs:
           key: v1-dependency-cache-{{ checksum "st2/requirements.txt" }}
       - run: sudo apt install python-dev
       - run: make ewcdocs
+      - run:
+          name: Store HTML docs in workspace dir
+          command: mkdir /tmp/ewcdocs && cp -r docs/build/html/* /tmp/ewcdocs/
+      - persist_to_workspace:
+          root: /tmp
+          paths:
+            - ewcdocs
       - save_cache:
           key: v1-dependency-cache-{{ checksum "st2/requirements.txt" }}
           paths:
             - ~/.cache/pip
+  deploy:
+    docker:
+      - image: cibuilds/aws:1.16.43
+    steps:
+      - attach_workspace:
+          at: ./generated-site
+      - run:
+          name: Deploy to docs.stackstorm.com and ewc-docs.extremenetworks.com
+          command: |
+            if [ "${CIRCLE_BRANCH}" = "master" ]; then
+              aws s3 sync generated-site/st2docs/ \
+              s3://${ST2DOCS_BUCKET}/latest
+              aws s3 sync generated-site/ewcdocs/ \
+              s3://${EWCDOCS_BUCKET}/latest
+            else
+              aws s3 sync generated-site/st2docs/ \
+              s3://${ST2DOCS_BUCKET}/${CIRCLE_BRANCH}
+              aws s3 sync generated-site/ewcdocs/ \
+              s3://${EWCDOCS_BUCKET}/${CIRCLE_BRANCH}
+            fi
+      - run:
+          # Check the install scripts to see if this is the current GA version
+          name: Check if current GA branch, in which case also deploy to main site
+          command: |
+            GA_VER=$(curl -sSL https://stackstorm.com/packages/install.sh|grep ^BRANCH=|sed 's/[^0-9.]*//g')
+            if [ "${CIRCLE_BRANCH}" = "${GA_VER}" ]; then
+              aws s3 sync generated-site/st2docs/ \
+              s3://${ST2DOCS_BUCKET}/
+              aws s3 sync generated-site/ewcdocs/ \
+              s3://${EWCDOCS_BUCKET}/
+            else
+              echo "Not current GA version"
+            fi
+
 workflows:
   version: 2
-  docs_checks:
+  build-deploy:
     jobs:
       - st2docs
       - ewcdocs
+      - deploy:
+          requires:
+            - st2docs
+            - ewcdocs
+          filters:
+            branches:
+              only:
+                - master
+                - /v[0-9]+\.[0-9]+/
```

Comment:
I guess successful deployment will be a best reviewer.
You can temporary add `docs_build` branch to the allowed list to verify if deployment really works as expected.

Reply:
Yep, I tried that - I've removed it from the final PR, but I did that during my tests. Worked as expected (once I fixed a couple of bugs :))